stable/database/templates/_helpers.tpl

```diff
@@ -649,4 +649,92 @@ calculated and appended to the truncated string to make it unique.
 {{- else -}}
 {{ $s }}
 {{- end -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{/*
+Renders the TLS secrets as projected volume. Combining all secrets into one
+volume allows mounting them into single directory (/etc/nuodb/keys) without
+"subPath".
+*/}}
+{{- define "database.tlsVolume" -}}
+{{- if .Values.admin.tlsClientPEM }}
+- name: tls
+  projected:
+    defaultMode: 0440
+    sources:
+{{- if .Values.admin.tlsCACert }}
+    - secret:
+        name: {{ .Values.admin.tlsCACert.secret }}
+        items:
+        - key: {{ .Values.admin.tlsCACert.key }}
+          path: ca.cert
+{{- end }}
+{{- if .Values.admin.tlsKeyStore }}
+    - secret:
+        name: {{ .Values.admin.tlsKeyStore.secret }}
+        items:
+        - key: {{ .Values.admin.tlsKeyStore.key }}
+          path: nuoadmin.p12
+{{- end }}
+{{- if .Values.admin.tlsClientPEM }}
+    - secret:
+        name: {{ .Values.admin.tlsClientPEM.secret }}
+        items:
+        - key: {{ .Values.admin.tlsClientPEM.key }}
+          path: nuocmd.pem
+{{- end }}
+{{- end }}
+{{- end -}}
+
+{{/*
+Renders the TLS password for keystore or truststore.
+*/}}
+{{- define "database.keystorePassword" -}}
+{{- $ := index . 0 -}}
+{{- $store := index . 1 -}}
+{{- if $store.password -}}
+{{ $store.password }}
+{{- else if $store.passwordKey -}}
+  {{- $secret := lookup "v1" "Secret" $.Release.Namespace $store.secret -}}
+  {{- if $secret -}}
+  {{- $encoded := index $secret "data" $store.passwordKey -}}
+    {{- if $encoded -}}
+{{ print $encoded | b64dec }}
+    {{- end -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Renders the TLS keystore or truststore.
+*/}}
+{{- define "database.keystore" -}}
+{{- $ := index . 0 -}}
+{{- $store := index . 1 -}}
+{{- $secret := lookup "v1" "Secret" $.Release.Namespace $store.secret -}}
+{{- if $secret -}}
+  {{- $encoded := index $secret "data" $store.key -}}
+  {{- if $encoded -}}
+{{ print $encoded | b64dec }}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Renders TE/SM Pod annotations for TLS checksum that forces engine pods restart on
+configuration change. Pod restart is needed if the keystore or its password
+have changed. If the AP server certificate is rotated, new process certificates
+will be generated by NuoAdmin on engine startup.
+*/}}
+{{- define "database.tlsConfigAnnotations" -}}
+{{- if .Values.admin.tlsKeyStore }}
+  {{- $password := include "database.keystorePassword" (list . .Values.admin.tlsKeyStore) -}}
+  {{- if $password }}
+checksum/tls-passwords: {{ sha256sum $password }}
+  {{- end }}
+  {{- $keystore := include "database.keystore" (list . .Values.admin.tlsKeyStore) -}}
+  {{- if $keystore }}
+checksum/tls-keystore: {{ sha256sum $keystore }}
```

Comment:
The user still has to take manual action for the new keystore and truststore to be loaded. Before it was by running `kubectl rollout statefulset/deployment ...`, and now it is by running `helm upgrade ...` just to cause the checksum to be recalculated from the secret resources.